src/rluni/robot/robot.py

```diff
@@ -0,0 +1,428 @@
+        self.imu_config = pkg_resources.resource_filename("rluni", self.imu_config)
+
+        self.rlmodel_path = gvcv(
+            config, "RobotSystem.rlmodel_path", str, required=False
+        )
+        self.rlmodel_path = pkg_resources.resource_filename("rluni", self.rlmodel_path)
+
+        self.pid_config_path = gvcv(
+            config, "RobotSystem.pid_config", str, required=False
+        )
+        self.pid_config_path = pkg_resources.resource_filename(
+            "rluni", self.pid_config_path
+        )
+
+    def _get_controller(self, controller_type: str) -> Controller:
+        """Initialize the controller based on the type ('pid', 'rl', 'lqr', 'test')."""
+        if controller_type == "pid":
+            return PIDController(config_file=self.pid_config_path)
+        elif controller_type == "rl":
+            return RLController(model_pth=self.rlmodel_path)
+        elif controller_type == "lqr":
+            return LQRController()
+        elif controller_type == "test":
+            return TestController()
+        else:
+            raise ValueError(f"Unsupported controller type: {controller_type}")
+
+    async def start(self, shutdown_event):
+        """
+        Starts the control loop.
+        """
+        try:
+            await self.control_loop(shutdown_event)
+        except asyncio.CancelledError:
+            logger.info("Robot Loop Shutdown Signal.")
+            # Set the shutdown event to ensure control_loop exits
+            shutdown_event.set()
+            raise
+
+    async def control_loop(self, shutdown_event):
+        """
+        The main control loop for the robot. Executes sensor reading, state estimation, control decision making,
+        and actuator commands at a fixed rate defined by LOOP_TIME.
+        """
+        loop_period = self.LOOP_TIME
+        torque_request = 0
+
+        while not shutdown_event.is_set():
+            # Start Loop Timer and increment loop iteration
+            loop_start_time = time.time()
+            self.itr += 1
+
+            # Debugging Data Example (add more data as needed)
+            tele_debug_data = td.DebugData()
+            tele_debug_data.add_data(example_debug_data=42)
+
+            # Sensor reading and fusion
+
+            imudata = td.IMUData(*self.imu.read_sensor_data(convert=True))
+
+            quaternion, internal_states, flags = self.sensor_fusion.update(
+                imudata.get_gyro(),
+                imudata.get_accel(),
+                mag_data=imudata.get_mag(),
+                delta_time=loop_period,
+            )
+            rigid_body_state = td.EulerAngles(*self.sensor_fusion.euler_angles, *self.sensor_fusion.euler_rates)
+            
+            # Update robot state and parameters
+            # if self.xmotor is not None:
+            #     await self.xmotor.update_state()
+
+            for motor in self.motors:
+                if motor is not None: 
+                    await motor.update_state()
```

Comment:
Ahead we should see if this can be arranged by a single transport instead of 3 actions, fine for now (might be only way)